src/data/SettingsDataHandler.ts

```diff
@@ -0,0 +1,146 @@
+    private static instance: SettingsDataHandler | undefined = undefined;
+    private settings: Setting[] = [];
+    private unsavedChanges: boolean = false;
+    private listeners: (() => void)[] = [];
+    private idListeners: { iD: string; listener: () => {} }[] = [];
+
+    private constructor() {}
+
+    public hasUnchangedChanges() {
+        return this.unsavedChanges;
+    }
+
+    public static getInstance() {
+        if (SettingsDataHandler.instance === undefined) {
+            SettingsDataHandler.instance = new SettingsDataHandler();
+        }
+        return SettingsDataHandler.instance;
+    }
+
+    //initialize settings from server
+    public initializeSettings(data: Setting[]) {
+        data.forEach((setting) => {
+            setting.initialValue = setting.value;
+        });
+        this.settings = data;
+    }
+
+    public addListener(listener: () => void) {
+        if (!this.listeners.find((currentListener) => currentListener === listener)) {
+            this.listeners.push(listener);
+        }
+    }
+
+    removeListener(listener: () => void) {
+        this.listeners.filter((currentListener) => currentListener !== listener);
+    }
+
+    public addIdListener(listener: () => void, iD: string) {
+        if (!this.idListeners.find((element) => element.listener === listener)) {
+            this.listeners.push(listener);
+        }
+    }
+
+    removeIdListener(listener: () => void) {
+        this.idListeners.filter((element) => element.listener !== listener);
+    }
+
+    private callAllListeners() {
+        this.listeners.forEach((listener) => listener());
+    }
+
+    //TODO: save changes to server (batched)
+    public saveAll() {
+        this.settings.forEach(async (setting) => {
+            if (setting.initialValue !== setting.value) {
+                await this.saveSingleSetting(setting);
+            }
+        });
+        this.settings.forEach((setting) => (setting.initialValue = setting.value));
+        this.unsavedChanges = false;
+        this.callAllListeners();
+    }
+
+    private saveSingleSetting(setting: Setting) {
+        const api = new TeddyCloudApi(defaultAPIConfig());
+        const triggerWriteConfig = async () => {
+            await api.apiTriggerWriteConfigGet();
+        };
+
+        try {
+            return api
+                .apiPostTeddyCloudSetting(setting.iD, setting.value, setting.overlayId)
+                .then(() => {
+                    triggerWriteConfig();
+                    message.success(t("settings.saved") + ": " + setting.label);
+                })
+                .catch((e) => {
+                    message.error("Error while sending data to file: " + setting.label);
+                });
+        } catch (e) {
+            message.error("Error while sending data to server: " + setting.label);
+            return Promise<null>;
+        }
+
+        //TODO: what did this do in InputField? helpers.setValue(field.value || "");
```

Comment:
not sure, maybe it is just a fix that the IDE thinks field.value could be undefined and then only field.value leads to an error. To avoid that || "" was added.